### PR TITLE
Add Support for Email Options

### LINKED ISSUE #797: XML option to control email at end of the run
On slack, @rljacob said
> Some people want to turn off or on the "email at end of job" setting.

This is something I would like to be able to control, and I think `env_batch.xml` is the way to do it:

> I was thinking it would be nice to run `./xmlchange JOB_EMAIL_ON_COMPLETION=...` either as a binary or a string compare (`valid_values=always, if_success, if_failure,never`?)

I think the actual XML variable name and the best implementation is still open for debate.

### PR DESCRIPTION
**Pull request by erichlf**
Added an email option to env_batch.xml. Also, add the ability to send an email option through case.submit. The flags in case.submit follow slurm's usage, i.e. --mail-user and --mail-type, where type here is never, all, begin, end, fail.

Test suite: scripts_regression_tests.py (Anvil, Skybridge, and Climate)
various email options passed to case.submit on the above three machines, and similarly in env_batch.xml.
Test baseline: 
Test namelist changes:
Test status: bit for bit

Fixes #797

User interface changes?: Added --mail-user and --mail-type to case.submit

Update gh-pages html (Y/N)?: N

Code review: @jgfouca @jedwards4b

### CODE REVIEW

File: scripts/create_newcase [hunk 1]
```diff
@@ -171,7 +171,7 @@ def _main_func(description):
         user_mods_dir, user_compset, pesfile, \
         user_grid, gridfile, srcroot, test, ninst, walltime, queue, \
         output_root, script_root, run_unsupported, \
-        answer, input_dir = parse_command_line(sys.argv, cimeroot, description)
+        answer, input_dir, = parse_command_line(sys.argv, cimeroot, description)
```
Comment (jedwards4b) [inline scripts/create_newcase]: This addition seems wrong.

File: scripts/lib/CIME/XML/env_batch.py [hunk 2]
```diff
@@ -431,6 +450,69 @@ def _submit_single_job(self, case, job, depid=None, no_batch=False, skip_pnl=Fal
             logger.info("Submitted job id is {}".format(jobid))
             return jobid
 
+    def get_batch_mail_flag(self, batch_system):
```
Comment (jedwards4b) [inline scripts/lib/CIME/XML/env_batch.py]: These should be in data not in python.

File: scripts/lib/CIME/XML/env_batch.py [hunk 3]
```diff
@@ -431,6 +450,69 @@ def _submit_single_job(self, case, job, depid=None, no_batch=False, skip_pnl=Fal
...
+            'cobalt' : '-M',
+            'cobalt_theta' : '-M',
+            'pbs' : '-M',
+            'slurm' : '--mail-user',
+            'lsf' : '-u',
+        }
+
+        return mail_user_flag[batch_system]
+
+    def get_batch_mail_type(self, batch_system, mail_type='never'):
```
Comment (jedwards4b) [inline scripts/lib/CIME/XML/env_batch.py]: Should not be defined in python - should be in config_batch.xml

File: scripts/create_newcase (hunk 1 above)
Comment (erichlf) [inline scripts/create_newcase]: Not even sure how that got there. I don't remember doing that, but clearly I must have.

File: config/acme/machines/config_batch.xml [hunk 4]
```diff
@@ -77,6 +77,7 @@
       <arg flag="-q" name="$JOB_QUEUE"/>
       <arg flag="-W" name="$JOB_WALLCLOCK_TIME"/>
       <arg flag="-P" name="$PROJECT"/>
+      <arg flag="-u" name="$MAIL_USER"/>
```
Comment (jgfouca) [inline config/acme/machines/config_batch.xml]: Remove these. Instead add a <batch_email_flag> attribute to the batch system.

Example
<batch_email_flag>u</batch_email_flag>

File: config/acme/machines/config_batch.xml [hunk 5]
```diff
@@ -100,13 +101,14 @@
       <arg flag="-q" name="$JOB_QUEUE"/>
       <arg flag="-l walltime=" name="$JOB_WALLCLOCK_TIME"/>
       <arg flag="-A" name="$PROJECT"/>
+      <arg flag="-M" name="$MAIL_USER"/>
+      <arg flag="-m" name="$MAIL_TYPE"/>
```
Comment (jgfouca) [inline config/acme/machines/config_batch.xml]: Remove these. Instead add a <batch_email_type_flag> attribute to the batch system.

Example:
<batch_email_type_flag>, aeb, b, e, a</batch_email_type_flag>

File: scripts/lib/CIME/XML/env_batch.py (hunk 2 above)
Comment (jgfouca) [inline scripts/lib/CIME/XML/env_batch.py]: Yes, should be:

```
self.get_value("batch_email_flag")
```

File: scripts/lib/CIME/XML/env_batch.py (hunk 3 above)
Comment (jgfouca) [inline scripts/lib/CIME/XML/env_batch.py]: Yes, should be:

```
raw =  self.get_value("batch_email_type_flag")
items = [item.strip() for item in raw.split(",")]
idx = ["never", "all", "begin", "end", "fail"].index(mail_type)
return items[idx]
```

**Comment by erichlf** [comment]
I am testing things, so don't merge yet.

**Comment by rljacob** [comment]
Erich, are you using xmllint?

**Comment by jedwards4b** [comment]
@rljacob I've confirmed that he used xmllint and updated the schema with the additional fields.
@erichlf Thanks for adding these changes to the cesm side as well.

**Comment by rljacob** [comment]
Ok.  I'd suggest re-writing the history of this branch to remove some of the typo commits.

**Comment by erichlf** [comment]
There is still at least one issue that I am working out, so don't merge this yet.

**Comment by erichlf** [comment]
Okay, things should be finally working.

**Pull request merged**